rewriter/rewriter.cc

```diff
@@ -185,6 +186,11 @@ class Rewriterer {
             return expr;
         }
 
+        if (auto expr = ChainedSig::run(ctx, send)) {
+            SigRewriter::run(ctx, ast::cast_tree<ast::Send>(expr));
+            return expr;
+        }
+
```

Comment:
I think this transformation happens to work by accident right now. I think that what we really want is to translate the new syntax to the old syntax as early as possible, namely: before any other rewriter passes fire.

The treemap where you've written this pass interleaves all the rewriter passes at the same time in a single traversal over the tree. That means that depending on a somewhat arbitrary ordering, portions of other rewriter passes could fire before this `ChainedSig` pass runs, which could then observe the new syntax instead of the old syntax.

What do you think about structuring the `ChainedSig` pass similar to the `Cleanup` pass, which manages it's own treemap, and then running `ChainedSig::run` before the `TreeMap::apply` on these lines?

https://github.com/sorbet/sorbet/blob/a9fb984be106118b7f5b757c29f09b70993e8792/rewriter/rewriter.cc#L202-L203
There are some rewriter passes that attempt to parse sigs. If this `ChainedSig` pass ran

Reply:
(That way, all the rewriter passes see a single, unified view of sigs when then attempt to parse them, and don't need to see multiple syntaxes. It also gives us a convenient leverage point to implement an autocorrect in the future.)

Reply:
Moved it to be an initial pass on the ast, like you mentioned.